spec/lib/rack/content_type_default_spec.rb

```diff
@@ -15,63 +17,117 @@
+      post_req['CONTENT_TYPE'] = ''
+      Rack::ContentTypeDefault.new(App).call(post_req)
+      post_req['CONTENT_TYPE'].should == 'application/json'
     end
 
-    it 'does not the content type for get request' do
-      request  = Rack::MockRequest.env_for("/api/v2/customers", method: :get)
-      Rack::ContentTypeDefault.new(App).call(request)
-      request['CONTENT_TYPE'].should_not == 'application/json'
+    it 'does not override the content type for get request' do
+      Rack::ContentTypeDefault.new(App).call(get_req)
+      get_req['CONTENT_TYPE'].should_not == 'application/json'
     end
   end
 
   context 'empty methods specified applies to no requests' do
     it 'does not set content type for any method' do
-      request  = Rack::MockRequest.env_for("/api/v2/customers", method: :post)
-      Rack::ContentTypeDefault.new(App, []).call(request)
-      request['CONTENT_TYPE'].should_not == 'application/json'
+      Rack::ContentTypeDefault.new(App, []).call(post_req)
+      post_req['CONTENT_TYPE'].should_not == 'application/json'
     end
   end
 
   context 'method' do
     it 'allows to specify the methods' do
-      request1  = Rack::MockRequest.env_for("/api/v2/customers", method: :get)
-      request2  = Rack::MockRequest.env_for("/api/v2/customers/1", method: :post)
-      Rack::ContentTypeDefault.new(App, :get).call(request1)
-      Rack::ContentTypeDefault.new(App, :get).call(request2)
-      request1['CONTENT_TYPE'].should == 'application/json'
-      request2['CONTENT_TYPE'].should_not == 'application/json'
+      Rack::ContentTypeDefault.new(App, :get).call(get_req)
+      Rack::ContentTypeDefault.new(App, :get).call(post_req)
+      get_req['CONTENT_TYPE'].should == 'application/json'
+      post_req['CONTENT_TYPE'].should_not == 'application/json'
     end
 
     it 'may be an array' do
-      request  = Rack::MockRequest.env_for("/api/v2/customers", method: :get)
-      Rack::ContentTypeDefault.new(App, [:get, :post]).call(request)
-      request['CONTENT_TYPE'].should == 'application/json'
+      Rack::ContentTypeDefault.new(App, [:get, :post]).call(get_req)
+      get_req['CONTENT_TYPE'].should == 'application/json'
     end
   end
 
   context 'content type' do
     it 'allows to specify the content type' do
-      request  = Rack::MockRequest.env_for("/api/v2/customers", method: :get)
-      Rack::ContentTypeDefault.new(App, :get, 'application/xml').call(request)
-      request['CONTENT_TYPE'].should == 'application/xml'
+      Rack::ContentTypeDefault.new(App, :get, 'application/xml').call(get_req)
+      get_req['CONTENT_TYPE'].should == 'application/xml'
     end
   end
-end
 
+  context 'paths' do
+    it 'sets the content type as appropriate if the path in the request matches one specified' do
+      get_req['PATH_INFO'] = 'api/v2/customers/2155/authenticate.json'
+      Rack::ContentTypeDefault.new(App, :get, 'application/xml', ['/show', '/authenticate.json']).call(get_req)
```

Comment:
This example is very interesting and not intuitive. thanks for adding this one!
so we are overriding the `.json` route's content-type to be `xml` :(

i hope is not a real use-case. which is why i asked the question on the main PR about your use-case.